### ISSUE TITLE
Comment character in editor for PRQL

### ISSUE THREAD

**Issue by richb-hanover** [timeseries/qstudio#68]
qStudio is working well with PRQL queries. Here's my next user experience "glitch":

Currently the qStudio editor inserts/removes "//" at the start of the line when someone hits Ctl-/ or Cmd-/

PRQL uses "#" to comment out lines. It would be great to detect `.prql` files and use the proper comment character/string. Many thanks.

**Comment by richb-hanover** [comment]
@ryanhamilton Thanks again for PRQL support in qStudio. This problem (wrong comment character) is a big blocker for me use qStudio further. The ability to comment out a line or two is critical to testing out PRQL queries. Thanks again.

**Comment by ryanhamilton** [comment]
Ah so this is a tricky one. 

1.If the file is named something.prql it will assume # as the comment marker. 
2. If you create a new unsaved file, it will guess based on the below code. Basically check what server we are connected to and assume the file is that type.

So the only way to guarantee a .prql comment works is to save the file.




```
 	String title = openDocumentsModel.getSelectedDocument().getTitle();
        	if(title.endsWith(".q")) {
        		return qFirst;
        	} else if(title.endsWith(".sql")) {
        		return sqlFirst;
        	} else if(title.endsWith(".prql")) {
        		return prqlFirst;
        	} else if(title.contains(".")) {
        		return anyFirst;
        	}
        	if(this.conMan.getServerConnections().isEmpty()) {
        		return qFirst;
        	}
        	ServerConfig sc = conMan.getServer(queryManager.getSelectedServerName());
        	return sc != null ? (JdbcTypes.DOLPHINDB.equals(sc.getJdbcType()) ? dosFirst : sc.isKDB() ? qFirst : sqlFirst) : qFirst;
```

Does that make sense?
Does it work when you save the file?

**Comment by richb-hanover** [comment]
It makes perfect sense to require the `.prql` suffix before qStudio changes its comment character. 

How can I get a version of qStudio to test? Thanks.

**Comment by ryanhamilton** [comment]
From the source code it looks like it was already in  since 3.02.
I just tested latest 3.06 and it definitely works.

**Comment by richb-hanover** [comment]
Hmmm... I created and saved the following in qStudio 3.06 as _test-of-comments.prql_ Hitting Cmd-/ on any of the in `select` lines on macOS gives a comment character of "//". 

AHAH! If I use **Ctl-/**, the comment character is a "#" as desired.

I recommend that the keyboard command for commenting a line be switched to Cmd-/ on macOS (Cmd-O opens a file, etc.)

Thanks!

```
prql target:sql.sqlite

from invoices
select {
    customer_id,
    invoice_date,
}
```

**Comment by ryanhamilton** [comment]
Should all of the below shortcuts use the command key in your opinion?
I don't have mac so I'm not sure.

```
Action.quick-find = jsyntaxpane.actions.QuickFindAction, control F
Action.find = jsyntaxpane.actions.FindReplaceAction, control H
Action.goto-line = jsyntaxpane.actions.GotoLineAction, control G
#Action.jindent = jsyntaxpane.actions.JIndentAction, control TAB
Action.toggle-comments = jsyntaxpane.actions.ToggleCommentsAction, control SLASH
```

**Comment by richb-hanover** [comment]
Yes - "Mac people" all instinctively go for the Command key when they use keyboard shortcuts. (Just as "Windows/Linux people" use Control.) I'm saying this as a 40-year Mac developer (Yikes! That's a long time!)

I checked all the keyboard shortcuts listed in the menus of qStudio, and they are (properly) listed as using Command (⌘) key. You should make sure that qStudio implements them all as shown (e.g., make sure that "Toggle comments" uses the Command key.)

I don't think it's critical, but if it would be helpful, I could produce a list of "Mac equivalents" of the most common keyboard shortcuts that are already in qStudio: Close window, Close all, Save as..., Print, Exit (Quit in macOS), Find, etc. I could also give Mac-isms for some of the current keyboard shortcuts that rely on the function keys (e.g., F3) since Mac users rarely use function keys in daily use. Thanks as always

**Comment by richb-hanover** [comment]
PS The Ctl-/ keyboard shortcut toggles comments as desired, but it also beeps. Could you silence that? Thanks.

**Comment by ryanhamilton** [comment]
I've tried changing the above shortcuts to command. It's released now in 3.07:
https://www.timestored.com/qstudio/download
Please confirm if possible it works.

**Comment by richb-hanover** [comment]
I can confirm that qStudio 3.07 _does_ toggle comments when typing Cmd-/. I checked several of the others and they worked as expected.

Let me know if it would be interesting to get a list of other common keyboard shortcuts on the Mac to make qStudio "feel more native". Thanks.
